nvflare/private/fed/server/job_cmds.py

```diff
@@ -407,12 +407,20 @@ def abort_job(self, conn: Connection, args: List[str]):
         try:
             job_id = conn.get_prop(self.JOB_ID)
             with engine.new_context() as fl_ctx:
-                message = job_runner.stop_run(job_id, fl_ctx)
-                if message:
-                    conn.append_error(message, meta=make_meta(MetaStatusValue.INTERNAL_ERROR, message))
-                else:
-                    conn.append_string("Abort signal has been sent to the server app.")
+                job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
+                job = job_manager.get_job(job_id, fl_ctx)
+                if job.meta.get(JobMetaKey.STATUS) in [RunStatus.SUBMITTED, RunStatus.DISPATCHED]:
+                    job_manager.set_status(job.job_id, RunStatus.FINISHED_ABORTED, fl_ctx)
```

Comment:
will we have other threads calling job_manager.set_status to the same job at the same time?

Reply:
we do have other places update the job status as well in JobRunner. We already add the protection to handle this as much as we can.